fbw-common/src/systems/instruments/src/remote-client.ts

```diff
@@ -0,0 +1,459 @@
+import { Wait } from '@microsoft/msfs-sdk';
+import { ClientState, NXDataStore, SimBridgeClientState } from '@flybywiresim/fbw-sdk';
+import { protocolV0 } from '@flybywiresim/remote-bridge-types';
+import { Base64 } from 'js-base64';
+
+import { v4 } from 'uuid';
+
+const MAX_CONNECTION_ATTEMPTS = 10;
+
+const EXCLUDED_DATA_STORAGE_KEYS = [
+    'A32NX_NAVIGRAPH_REFRESH_TOKEN',
+];
+
+interface SubscriptionDescriptor {
+    subscriptionID: string,
+    subscriptionGroupID: string,
+    fromClientID: string,
+    cancel(): void;
+}
+
+export interface RemoteClientOptions {
+    /** The WebSocket URL the remote client should connect to */
+    websocketUrl: () => string,
+
+    /** The name of the client */
+    clientName: string,
+
+    /** The name of the airframe the client is loaded into */
+    airframeName: string,
+
+    /** The VFS path of the instruments metadata file */
+    instrumentsMetadataFile: string,
+
+    /** The path which all file download requests must have at the start of their VFS path */
+    fileDownloadBasePath: string,
+}
+
+/**
+ * Client for connecting a gateway supporting the FBW remote bridge protocol
+ */
+export class RemoteClient {
+    private static readonly PROTOCOL_VERSION = 0;
+
+    private ws: WebSocket | null = null;
+
+    private readonly url: string;
+
+    private readonly airframeName: string;
+
+    private readonly clientName: string;
+
+    private readonly fileDownloadBasePath: string;
+
+    private readonly clientID = v4();
+
+    private instruments: protocolV0.InstrumentMetadata[] = [];
+
+    private connectionAttemptCount = 0;
+
+    private simVarSubscriptions: {
+        simVar: string,
+        simVarType: 'simVar' | 'globalVar',
+        unit: string,
+        id: number,
+        fromClientID: string,
+        subscriptionGroupID: string,
+    }[] = [];
+
+    private simVarValueCache: Record<number, any> = {};
+
+    private viewListeners = new Map<string, ViewListener.ViewListener>();
+
+    private aircraftDataUpdateInterval: number | null = null;
+
+    private subscriptions: SubscriptionDescriptor[] = [];
+
+    constructor(options: RemoteClientOptions) {
+        this.url = options.websocketUrl();
+        this.airframeName = options.airframeName;
```

Comment:
```suggestion
        this.airframeName = options.airframeName;
```

Maybe this should be aircraft instead of airframe (a bit pendatic but sure)